src/client/Client.cc

```diff
@@ -9243,14 +9243,18 @@ void Client::seekdir(dir_result_t *dirp, loff_t offset)
 //};
 void Client::fill_dirent(struct dirent *de, const char *name, int type, uint64_t ino, loff_t next_off)
 {
-  strncpy(de->d_name, name, 255);
-  de->d_name[255] = '\0';
+  int len = strlen(name);
+  if (len > 255) {
+    len = 255;
+  }
```

Comment:
```suggestion
  size_t len = strlen(name);
  len = min(len, (size_t)255);
```
`int` could get prone to overflow, therefore switched to using `size_t` (`memcpy()`'s len arg is indeed of type `size_t` too) and using `min()` looks a bit cleaner.

Reply:
It's very unlikely that it will overflow, but anyway size_t is better. I'll change this.